xbmc/pvr/epg/Epg.cpp

```diff
@@ -304,7 +304,7 @@ bool CPVREpg::Persist(const std::shared_ptr<CPVREpgDatabase>& database, bool bQu
 
   if (m_iEpgID <= 0 || m_bChanged)
   {
-    const int iId = database->Persist(*this, m_iEpgID > 0);
+    const int iId = database->Persist(*this);
```

Comment:
Not using the value `bQueueWrite` here was there for a reason. I doubt that just ignoring the value of `m_iEpgID` now is a good idea.

Reply:
IIRC correctly, it was very important that the first insert for an EPG (identified by m_iEpgId being less or equal zero) was done not queued, because the db generates the EPG id (`m_pDS->lastinsertid()`.

https://github.com/xbmc/xbmc/blob/master/xbmc/pvr/epg/EpgDatabase.cpp#L997

Reply:
No. Thank you for pointing out the requirement.

Reply:
I tried to test the `m_iEpgID <= 0` case but no avail.  A valid m_iEpgID seems ever be created when needed.

Do you have an idea how to force the condition (from normal operation)?